src/System.Device.Gpio/System/Device/I2c/I2cDevice.cs

```diff
@@ -70,7 +70,7 @@ public static I2cDevice Create(I2cConnectionSettings settings)
             }
             else
             {
-                return new UnixI2cDevice(settings);
+                return new UnixI2cDevice(UnixI2cBus.Create(settings.BusId), settings.DeviceAddress, shouldDisposeBus: true);
```

Comment:
Perhaps we should just deprecate this API in favor of the bus managing the creation of devices, what do you think?

Reply:
No, I think you should use I2cDevice.Create when you have one device and I2cBus when you have multiple I2C devices

Reply:
Basically we should ask ourselves this question: What is the point of having I2cDevice class after we introduce I2cBus the way you have it on this PR? The only benefit I see is so that you don't have to remember the I2cAddress, which to me makes very little of an argument to keep it. I'm not against having I2cBus having all t hese methods, but if that's the case then perhaps we should just deprecate the whole I2cDevice if we don't see any benefit of it other than remembering which address it's plugged into.

Reply:
I kind of think of this the same as XML APIs, they have different purpose even though you can achieve the same things with them: there is XmlDocument which is optimized for DOM manipulation, XDocument optimized for quick extraction of the data and XPathDocument optimized for making complex queries. Here I2cDevice is for manipulating on a single address, I2cBus is giving you way of creating devices and operating on the spectrum. Once we remove Read/Write/ReadWrite methods there should be easier distinction but I still think there is a value in having it in both places. Also see discussion in here: https://github.com/dotnet/iot/issues/214 the device mention in there uses different I2C address depending on the size of the memory.

Reply:
I2cDevice makes it easy when you have only 1 device. Similar somehow with GpioPin vs GpioController. What's the purpose of having a controller when you need only 1 pin? Same for me on the I2cDevice. 
Bindings so far are using only I2cDevice and it's good. We should continue to encourage this and continue to push for this except where there are multiple devices like sense hat, where reading and writing on the bus to initialize, send alerts (SMBus scenarios) is needed.